### DISCUSSION TITLE
Bill of materials

### DISCUSSION THREAD

**Discussion by DavidPHirsch** [todogroup/ospology#34] [OSPO Discussions]
Hi Everyone! How do you manage the Bill of Materials for open-source projects? do you create separate files in the repositories where you store the information? Are there any best practices you could point me to?
Thanks

**Comment by lightoyou** [comment]
Use [dependecy track](https://dependencytrack.org/) it's a nice tools that use https://cyclonedx.org/ standard and https://spdx.org/licenses/

**Reply by DavidPHirsch** [reply]
great, thank you! and where do you usually store the SBOM in a repo?

**Reply by stevespringett** [reply]
Usually alongside the artifact itself. Future versions of Dependency Track will bundle the [CycloneDX BOM Repository Server](https://github.com/CycloneDX/cyclonedx-bom-repo-server)

**Comment by luhring** [comment]
(I'm a [Syft](https://github.com/anchore/syft) maintainer.)

We believe a best practice is to store the SBOM **as close as possible to where you're publishing the artifact** that the SBOM describes.

For example, if you publish _container images_, we believe in putting SBOMs in the container registry, alongside the images themselves. ([Cosign](https://github.com/SigStore/cosign) makes this really easy.)

Another case we see a lot is shipping artifacts via _GitHub releases_. We make a [GitHub action](https://github.com/anchore/sbom-action/) that can automatically generate an SBOM and attach it to your release for you.

**Reply by DavidPHirsch** [reply]
great, thank you! and where do you usually store the SBOM in a repo?

**Reply by luhring** [reply]
Actually, I usually don't commit SBOMs into repos (instead favoring those other locations). Is that something you're seeing a need for specifically?

**Reply by jthDEV** [reply]
What about the idea to provide them via API. Thus you might add an identifier with your code and call the linked documentation through that API?

**Reply by stevespringett** [reply]
@jthDEV CycloneDX has a standard API supporting XML, JSON, and Protobuf that takes advantage of features of the spec, such as serial numbers and bom versions to publish and retrieve boms. It's currently implemented in https://github.com/CycloneDX/cyclonedx-bom-repo-server but we're working on a public exchange API that every repository server could theoretically implement.

**Reply by jthDEV** [reply]
Hi @stevespringett, thank you for the link. This looks good. We have provided a similar solution but we use an abstract UUID to address a specific BOM version. Probably you might want to consider this as well. The option to default back to the latest update bears the risk of mixup. The UUID may help to prevent this. Just as an idea.
